code/android-identity-library/src/main/java/com/adobe/marketing/mobile/identity/NetworkConnectionUtil.java

```diff
@@ -0,0 +1,117 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.identity;
+
+//TODO: kee this utility class in Identity package for now, move it to Core when used in other extensions.
+import com.adobe.marketing.mobile.util.StringUtils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.util.*;
+final class NetworkConnectionUtil {
+	private static final String DEFAULT_CHARSET = "UTF-8";
+	private static final int    BUF_SIZE        = 1024;
+
+	static final String HTTP_HEADER_KEY_CONTENT_TYPE = "Content-Type";
+	static final String HTTP_HEADER_KEY_ACCEPT_LANGUAGE = "Accept-Language";
+	static final String HTTP_HEADER_KEY_ACCEPT = "Accept";
+
+
+	static final String HTTP_HEADER_CONTENT_TYPE_JSON_APPLICATION = "application/json";
+	static final String HTTP_HEADER_CONTENT_TYPE_WWW_FORM_URLENCODED =
+		"application/x-www-form-urlencoded";
+	static final String HTTP_HEADER_ACCEPT_TEXT_HTML = "text/html";
+
+	private NetworkConnectionUtil() {}
+
+	static ArrayList<Integer> recoverableNetworkErrorCodes = new ArrayList<Integer>(Arrays.asList(
+				HttpURLConnection.HTTP_CLIENT_TIMEOUT,
+				HttpURLConnection.HTTP_GATEWAY_TIMEOUT,
+				HttpURLConnection.HTTP_UNAVAILABLE
+			));
+
+	/**
+	 * Calls {@link #readFromInputStream(InputStream, String)} with the default charset - UTF-8
+	 *
+	 * @param inputStream connection's input stream
+	 * @return data from input stream as string
+	 * @throws IOException if reading from input stream or conversion to string fails
+	 */
+	static String readFromInputStream(final InputStream inputStream) throws IOException {
+		return readFromInputStream(inputStream, DEFAULT_CHARSET);
+	}
+
+	/**
+	 * Reads data from the provided inputStream in chunks of 1024 and returns it as string. It uses charsetName UTF-8
+	 * for conversion to string
+	 *
+	 * @param inputStream connection's input stream
+	 * @param charset     optional parameter for charset, if it is missing UTF-8 constant will be used
+	 *
+	 * @return data from input stream as {@code String}
+	 * @throws IOException if reading from input stream or conversion to string fails
+	 */
+	static String readFromInputStream(final InputStream inputStream, final String charset) throws IOException {
```

Comment:
We expose similar function from Core 
https://github.com/adobe/aepsdk-core-android/blob/dev-v2.0.0/code/android-core-library/src/main/java/com/adobe/marketing/mobile/util/StreamUtils.java#L23